Source/Session.swift

```diff
@@ -442,6 +471,16 @@ open class Session {
             request.didSuspend()
         }
     }
+
+    // MARK: - Adapters and Retriers
+
+    func adapter(for request: Request) -> RequestAdapter? {
+        return request.interceptor ?? adapter
+    }
+
+    func retrier(for request: Request) -> RequestRetrier? {
+        return request.interceptor ?? retrier
+    }
```

Comment:
Nice and simple. Use the request's interceptor if specified, and fall back to the session if necessary.